lib/axon/compiler.ex

```diff
@@ -672,19 +707,9 @@ defmodule Axon.Compiler do
             {layer_inputs, rest, [param | inputs]}
         end)
 
-      # TODO: Hack for dropout with key, fix with a better implementation
-      opts =
-        if op in @dropout_layers do
-          <<data::unsigned-size(32), _rest::binary>> = :erlang.md5(name)
-          dropout_key = Nx.Random.fold_in(key, data)
-          opts ++ [key: dropout_key]
-        else
-          opts
-        end
-
       # Compute arguments to be forwarded and ensure `:mode` is included
       # for inference/training behavior dependent functions
-      args = Enum.reverse(tensor_inputs) ++ [Keyword.put(opts, :mode, mode)]
+      args = Enum.reverse(tensor_inputs, [Keyword.put(opts, :mode, mode)])
```

Comment:
`Enum.reverse(list, tail)` is an efficient version of `Enum.reverse(list) ++ tail`.